src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTag.cs

```diff
@@ -254,5 +257,19 @@ private async Task StartToolTipServiceAsync(IToolTipPresenter toolTipPresenter)
 
             toolTipPresenter.StartOrUpdate(_textView.TextSnapshot.CreateTrackingSpan(_span.Start, _span.Length, SpanTrackingMode.EdgeInclusive), uiList);
         }
+
+        private void Adornment_MouseLeftButtonDown(object sender, MouseButtonEventArgs e)
+        {
+            if (e.ClickCount == 2)
+            {
+                e.Handled = true;
+                var replacementText = _hint.ReplacementText;
+                if (replacementText != null)
+                {
+                    var replacementValue = replacementText.Value;
+                    _ = _textView.TextBuffer.Replace(new VisualStudio.Text.Span(replacementValue.Span.Start, replacementValue.Span.Length), replacementValue.NewText);
```

Comment:
this code is likely wrong in any sort of progression scenario.  you're editing the topmost buffer, not the buffer containing c#.  we shuold test this in a legacy razor case.  @NTaylorMullen can likely help you with that.